### PR TITLE
EIP-1559 for substrate mainnet

### PR DESCRIPTION
**Pull request by akorchyn**
This pull request implements a modified version of the EIP-1559 standard for the substrate part with the following changes:
* Funds are transferred to the treasury instead of being burned.
* 25% of the priority fee (tips) is directed to the treasury.
* Tests have been added and refactored to ensure proper functionality.

### CODE REVIEW

**Comment by akorchyn** [comment]
image
Transfer 500 tokens to another account and include one ggx as a tip.
The base fee was transferred with 0.25 of the tip to the treasury.

File: Cargo.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ futures = "0.3.28"
 jsonrpsee = "0.16.2"
 log = "0.4.17"
 once_cell = { version = "1.17.1", default-features = false }
+paste = "1.0"
 proptest = "1.1"
```
Comment (xcthulhu) [inline Cargo.toml:28]: I think these are `dev-dependencies`...?

**Review by xcthulhu — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (akorchyn) [reply Cargo.toml:28]: You probably have a misunderstanding for workspace deps compared to crate deps

In the workspace, you state versioning and feature gates, then use whatever you want (deps, dev-deps) in crate Cargo.toml.

So I'm using it as a dev dependency in runtime/mainnet/Cargo.toml

File: runtime/mainnet/src/pos/currency.rs [hunk 2]
```diff
@@ -169,6 +179,18 @@ pub mod pallet {
 			Self::deposit_event(Event::TreasuryCommissionFromFeeChanged(new_commission));
 			Ok(())
 		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight(100_000)]
```
Comment (crawter) [inline runtime/mainnet/src/pos/currency.rs:184]: leave a TODO explaining why is it hardcoded here

**Review by crawter — APPROVED** [review]

**Pull request merged**